aws/resource_aws_acm_certificate.go

```diff
@@ -77,10 +85,18 @@ func resourceAwsAcmCertificate() *schema.Resource {
 }
 
 func resourceAwsAcmCertificateCreate(d *schema.ResourceData, meta interface{}) error {
+	rand.Seed(time.Now().UnixNano())
 	acmconn := meta.(*AWSClient).acmconn
 	params := &acm.RequestCertificateInput{
 		DomainName:       aws.String(d.Get("domain_name").(string)),
-		ValidationMethod: aws.String(d.Get("validation_method").(string)),
+		IdempotencyToken: aws.String(strconv.Itoa(rand.Int())),
+	}
+
+	caARN, ok := d.GetOk("certificate_authority_arn")
+	if ok {
+		params.CertificateAuthorityArn = aws.String(caARN.(string))
+	} else {
+		params.ValidationMethod = aws.String(d.Get("validation_method").(string))
```

Comment:
The changes made to move the `validation_method` argument from required to optional might require some extra validation to prevent calling the API for public certificate with the unsupported value `NONE` (if such an argument is not set).

Reply:
Addressed here https://github.com/micbase/terraform-provider-aws/blob/72d1fc49d8e2e5ad7b415f16e47c712608b70328/aws/resource_aws_acm_certificate.go#L123-L125